### DISCUSSION TITLE
Local devlopment setup for Horreum

### DISCUSSION THREAD

**Discussion by shivarm** [Hyperfoil/Horreum#347] [Q&A] [answered]
How can i run this project locally?

**Comment by jesperpedersen** [comment, marked as answer]
Currently, there is a description in the `README.md` to get it to run locally.

We should look into creating a distribution that - although depending on other services - is easier to get it running with. Is that something you are willing to work on ?

**Comment by shivarm** [comment]
hey @jesperpedersen  Horreum - Migrate project from React 4 to React 5, but this project has created using react 17. How should i start?

**Comment by jesperpedersen** [comment]
@shivam-sharma7 You are right - there are probably too many version numbers to discuss, but `17` to `18` is correct. The project will focus on using the latest libraries in the web stack such that `package.json` is current

**Comment by shivarm** [comment]
hi @jesperpedersen,

i want to ask you that would i need to open all `directory` to work on the webapp? or I can only navigate in webapp and open with VS Code. Because I don't think i need pom.xml file to build that

### Screenshort

![webapp](https://user-images.githubusercontent.com/91419219/216886366-b9e62371-8ff8-4f0c-903e-d548df0b1007.png)

**Comment by jesperpedersen** [comment]
The `webapp` directory is the primary directory for this work.

However, you need to be able to start and run the application. You can do that from the command line though.

**Reply by shivarm** [reply]
I tried hard but `npm start` not working. this is Only one problem to me  that i am facing in order to see website and how changes reflect on page

**Reply by jesperpedersen** [reply]
It would be nice to have it working, but the primary way of starting the application is through the Maven target

**Comment by shivarm** [comment]
hi,

when i made this following changes only in package.json then it started my local development server but not showing anything.

i got error. 

./src/api.tsx
Module not found: Can't resolve './generated' in 'C:\Users\shivam sharma\Documents\Gsoc-2023\Horreum\webapp\src'
---

![Screenshot_20230214_192900](https://user-images.githubusercontent.com/91419219/218759633-f6bb877e-8076-4b20-8343-2f37ce4ad829.png)

**Reply by jesperpedersen** [reply]
That is being done in the https://github.com/Hyperfoil/Horreum/blob/master/horreum-model/pom.xml#L98 step.

Since these files will always be generated, and therefore ignored in Git you either need to run the "full" compile, or generate them once and keep them around locally. However, the model will change as the project progress.
